contrib/cai-mock-data-generator/tests/generator_test.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2019 The Forseti Security Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Generator Tests."""
+
+import collections
+import json
+import pathlib
+import unittest
+
+from data_generator import generator
+from data_generator.file_handler import read_yaml_file
+from tests.resource_node import ResourceNode
+
+
+class TestGenerator(unittest.TestCase):
+
+    file_paths = ['{}/{}-resource.dump', '{}/{}-iam-policy.dump']
+
+    def tearDown(self) -> None:
+        """Remove all the dump files."""
+        for path in pathlib.Path('.').glob('*.dump'):
+            path.unlink()
+
+    def test_count_folders(self):
+        config_file_path = './config/config_100_folders.yaml'
+        config = read_yaml_file(config_file_path)
+        generator.CAIDataGenerator(config).generate()
+        paths = list(map(lambda p: p.format(config.get('output_path'),
+                                            config.get('output_file_name_prefix')),
+                     self.file_paths))
+        root_cai_name = '//cloudresourcemanager.googleapis.com/organizations/{}'.format(
+            config.get('root_resource_id'))
+        expected_resource_node = ResourceNode(root_cai_name, 'cloudresourcemanager.googleapis.com/Organization')
+        expected_resource_node.children = [ResourceNode(i, 'cloudresourcemanager.googleapis.com/Folder') for i in range(100)]
+
+        resource_count, root_resource_node = self.count_resource_by_type(paths[0])
+        iam_count, _ = self.count_resource_by_type(paths[1])
+        self.assertEqual(expected_resource_node, root_resource_node)
+        self.assertEqual(1, resource_count.get('cloudresourcemanager.googleapis.com/Organization', 0))
+        self.assertEqual(1, iam_count.get('cloudresourcemanager.googleapis.com/Organization', 0))
+        self.assertEqual(100, resource_count.get('cloudresourcemanager.googleapis.com/Folder', 0))
+        self.assertEqual(100, iam_count.get('cloudresourcemanager.googleapis.com/Folder', 0))
+
+    def test_count_folders_projects(self):
+        config_file_path = './config/config_folders_projects.yaml'
+        config = read_yaml_file(config_file_path)
+        generator.CAIDataGenerator(config).generate()
+        paths = list(map(lambda p: p.format(config.get('output_path'),
```

Comment:
since this is being re-used, it should be refactored out into a helper method.

Reply:
a small test for the helper method would be great

Reply:
Break it up to not use map instead